File: clear_db.js
const https = require('https');

const projectId = 'dbros-apps-7bbmw4';
const baseUrl = `https://firestore.googleapis.com/v1/projects/${projectId}/databases/(default)/documents`;

function request(url, options = {}) {
  return new Promise((resolve, reject) => {
    const req = https.request(url, options, (res) => {
      let data = '';
      res.on('data', chunk => data += chunk);
      res.on('end', () => {
        if (res.statusCode >= 200 && res.statusCode < 300) {
          resolve(data ? JSON.parse(data) : null);
        } else {
          reject(new Error(`HTTP ${res.statusCode}: ${data}`));
        }
      });
    });
    req.on('error', reject);
    req.end();
  });
}

async function deleteAll(collection) {
  try {
    const data = await request(`${baseUrl}/${collection}`);
    const docs = data && data.documents ? data.documents : [];
    
    if (docs.length === 0) {
      console.log(`[${collection}] No documents found.`);
      return;
    }
    
    console.log(`[${collection}] Found ${docs.length} documents to delete.`);
    for (const doc of docs) {
      console.log(`Deleting ${doc.name}...`);
      await request(`https://firestore.googleapis.com/v1/${doc.name}`, { method: 'DELETE' });
    }
    console.log(`[${collection}] All documents deleted.`);
  } catch (e) {
    console.error(`Error with ${collection}:`, e.message);
  }
}

async function main() {
  await deleteAll('notices');
  await deleteAll('admin_push_requests');
}

main();

File: clear_db.py
import urllib.request
import urllib.error
import json

project_id = 'dbros-apps-7bbmw4'
base_url = f'https://firestore.googleapis.com/v1/projects/{project_id}/databases/(default)/documents'

def delete_all(collection):
    url = f'{base_url}/{collection}'
    try:
        req = urllib.request.Request(url)
        with urllib.request.urlopen(req) as response:
            data = json.loads(response.read().decode())
            docs = data.get('documents', [])
            
            if not docs:
                print(f"[{collection}] No documents found.")
                return
                
            print(f"[{collection}] Found {len(docs)} documents to delete.")
            for doc in docs:
                doc_name = doc['name']
                print(f"Deleting {doc_name}...")
                del_req = urllib.request.Request(f"https://firestore.googleapis.com/v1/{doc_name}", method='DELETE')
                urllib.request.urlopen(del_req)
            print(f"[{collection}] All documents deleted.")
            
    except urllib.error.HTTPError as e:
        print(f"Error fetching {collection}: {e}")

delete_all('notices')
delete_all('admin_push_requests')
